pylint/test/functional/keyword_arg_before_vararg.py

```diff
@@ -0,0 +1,35 @@
+"""Unittests for W1125 (kw args before *args)"""
+from __future__ import absolute_import, print_function
+
+# pylint: disable=unused-argument
+def check_kwargs_before_args(param1, param2=2, *args):
+    """docstring"""
+    pass
+
+check_kwargs_before_args(5) # [keyword-arg-before-vararg]
+
+# pylint: disable=too-few-public-methods, invalid-name
+class AAAA(object):
+    """class AAAA"""
+    def func_in_class(self, param1, param2=2, *args):
+        "method in class AAAA"
+        pass
+
+    @staticmethod
+    def static_method_in_class(param1, param2=3, *args):
+        "static method in class AAAA"
+        pass
+
+    @classmethod
+    def class_method_in_class(cls, param1, param2=4, *args):
+        "class method in class AAAA"
+        pass
+
+some_var = AAAA()
+some_var.func_in_class(3) # [keyword-arg-before-vararg]
+
+some_var.static_method_in_class(4) # [keyword-arg-before-vararg]
```

Comment:
I think I omitted this last time I reviewed this PR. This warning should be emitted when the function is defined, not when the function is called, since the problem is with the function definition, not with its application. Look for a `visit_functiondef` in that file and emit the message there.

Reply:
Agree with you. Have put the logic into design_analysis.py which I feel is the right place now. There was no 'visit_functiondef' function in the typecheck.py.